Lib/multiprocessing/connection.py

```diff
@@ -372,11 +373,14 @@ def _close(self, _close=_multiprocessing.closesocket):
             _close(self._handle)
         _write = _multiprocessing.send
         _read = _multiprocessing.recv
+        _default_pipe_size = 0
     else:
         def _close(self, _close=os.close):
             _close(self._handle)
         _write = os.write
         _read = os.read
+        _base_page_size = os.sysconf(os.sysconf_names['SC_PAGESIZE'])
+        _default_pipe_size = _base_page_size * 16
```

Comment:
First time commenter here,

Why 16?

(sorry for butting in 🙏)

Reply:
The default size of a pipe on Linux systems is 16 times the base page size: 
https://man7.org/linux/man-pages/man7/pipe.7.html

Reply:
That is a good question, and one that may come up for people reading this code. I suggest adding a module level constant (PIPE_CAPACITY? PIPE_MAX_PAGES?) and a comment pointing to that link. This way, users that have different configurations for pipe sizes can adjust the _default_pipe_size by changing the constant.

Reply:
I see, thanks.
Can that reply be inlined as a comment or as a constant? (assuming it's generally acceptable in the codebase)

Reply:
Is it okay to add such a constant just above the `clas Connection`, or is there a specific file, where constants are put?

Reply:
I am not sure this complexity is worth enough, compared to just hardcode 64KiB.

Does all platform except Windows supports `os.sysconf(os.sysconf_names['SC_PAGESIZE'])`?
Does they all use 16 pages?

Reply:
Hardcoding the value would lead to performance degradation for systems that use base page sizes larger than 4 KiB, e.g. like possible on ARM64, where you can configure base pages to have a size of  64 KiB. 

As for the other two questions, both BSD and macOS support the sysconf library function. As mentioned below, the size is the same on those systems:
https://www.netmeister.org/blog/ipcbufs.html

The main focus here is to avoid the creation of huge input buffers, which leads to a lot of costly system calls and huge page allocation.

Reply:
> Hardcoding the value would lead to performance degradation for systems that use base page sizes larger than 4 KiB, e.g. like possible on ARM64, where you can configure base pages to have a size of 64 KiB.

Is it realistic problem? How much degradation is expected?

> As for the other two questions, both BSD and macOS support the sysconf library function.

OK. I confirmed that is defined in POSIX.

> As mentioned below, the size is the same on those systems:
> https://www.netmeister.org/blog/ipcbufs.html

I think the "same" means 64KiB instead of 16 pages. Am I right?

https://opensource.apple.com/source/xnu/xnu-1504.9.37/bsd/sys/pipe.h

```
> #define BIG_PIPE_SIZE	(64*1024)
```

https://ftp.iij.ad.jp/pub/NetBSD/NetBSD-release-10/src/sys/sys/pipe.h

```
/*
 * Pipe buffer size, keep moderate in value, pipes take kva space.
 */
#ifndef PIPE_SIZE
#define PIPE_SIZE	16384
#endif

#ifndef BIG_PIPE_SIZE
#define BIG_PIPE_SIZE	(4*PIPE_SIZE)
#endif
```

> The main focus here is to avoid the creation of huge input buffers, which leads to a lot of costly system calls and huge page allocation.

Hardcoding 64KiB solves it too and simple. Only reason for having complex code is kernel-64k performance.

Reply:
I understand that writing complex code without a reason doesn't make much sense, but in this case, the pipe calculation covers the sizes of FreeBSD, macOS, and especially Linux. Furthermore, it covers cases of other CPUs that don't use 4 KiB base pages. In my opinion that makes the "added complexity" worthwhile.

Reply:
> but in this case, the pipe calculation covers the sizes of FreeBSD, macOS, and especially Linux.

What *covers* mean? At least, macOS uses 64K but this PR uses 256K. And it seems significantly slower than 64K.

I used this script to quick test.
https://gist.github.com/methane/a6cb799704a11f2bb2f64b16c0b830cc

Result:
```
# 64k buffer
real	0m15.282s
user	0m9.264s
sys	0m15.350s

real	0m15.229s
user	0m9.320s
sys	0m15.296s

real	0m15.161s
user	0m9.300s
sys	0m15.254s

# 256k buffer
real	0m16.779s
user	0m9.449s
sys	0m16.911s

real	0m16.807s
user	0m9.407s
sys	0m16.920s

real	0m17.088s
user	0m9.421s
sys	0m17.272s
```

Reply:
It isn't clear to me what you are comparing to what there. I suggest only comparing the performance after this PR to the behavior before this PR. Tweaking buffer sizes further on specific platforms can be considered further fine tuning. Ideally we want the OS to _tell us_ the actual buffer size to use.

Reply:
_(and I suggest following up on the issue itself for more attention rather than buried in a merged PR where nobody else is likely to look)_

Reply:
Continue in #123559